pkg/codegen/main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	v10 "github.com/rancher/aks-operator/pkg/apis/aks.cattle.io/v1"
+	_ "github.com/rancher/wrangler-api/pkg/generated/controllers/apiextensions.k8s.io"
+	controllergen "github.com/rancher/wrangler/pkg/controller-gen"
+	"github.com/rancher/wrangler/pkg/controller-gen/args"
+	"github.com/rancher/wrangler/pkg/crd"
+	"github.com/rancher/wrangler/pkg/yaml"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func main() {
+	controllergen.Run(args.Options{
+		OutputPackage: "github.com/rancher/aks-operator/pkg/generated",
+		Boilerplate:   "pkg/codegen/boilerplate.go.txt",
+		Groups: map[string]args.Group{
+			"aks.cattle.io": {
+				Types: []interface{}{
+					v10.AKSClusterConfig{},
+				},
+				GenerateTypes: true,
+			},
+			// Optionally you can use wrangler-api project which
+			// has a lot of common kubernetes APIs already generated.
+			// In this controller we will use wrangler-api for apps api group
+			"": {
+				Types: []interface{}{
+					v1.Pod{},
+					v1.Node{},
+					v1.Secret{},
+				},
+				InformersPackage: "k8s.io/client-go/informers",
+				ClientSetPackage: "k8s.io/client-go/kubernetes",
+				ListersPackage:   "k8s.io/client-go/listers",
+			},
+		},
+	})
+
+	aksClusterConfig := newCRD(&v10.AKSClusterConfig{}, func(c crd.CRD) crd.CRD {
+		c.ShortNames = []string{"akscc"}
+		return c
+	})
+
+	obj, err := aksClusterConfig.ToCustomResourceDefinition()
+	if err != nil {
+		panic(err)
+	}
+
```

Comment:
Why did the resource-policy get removed ? 
https://github.com/rancher/eks-operator/blob/master/pkg/codegen/main.go#L55

Reply:
I have backported `resource-policy` from eks